File: DSA/Madhav_590014266/Day29/Q2.java
class Solution {
    public String reverseString(String s) {
        Stack<Character> stack = new Stack<>();

        // Push all characters onto the stack
        for (char ch : s.toCharArray()) {
            stack.push(ch);
        }

        // Pop characters to build the reversed string
        StringBuilder result = new StringBuilder();
        while (!stack.isEmpty()) {
            result.append(stack.pop());
        }

        return result.toString();
    }
}

File: DSA/Aarush_590017879/Day_29/Q2.cpp
#include <iostream>
#include <stack>
#include <string>
using namespace std;

int main() {
    string s;
    cin >> s;

    stack<char> st;

    // Push all characters onto the stack
    for (char ch : s) {
        st.push(ch);
    }

    // Pop characters to form the reversed string
    string reversed = "";
    while (!st.empty()) {
        reversed += st.top();
        st.pop();
    }

    cout << reversed << endl;

    return 0;
}